aws/resource_aws_appautoscaling_scheduled_action.go

```diff
@@ -0,0 +1,222 @@
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"max_capacity": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+							ForceNew: true,
+						},
+						"min_capacity": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+			"schedule": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"start_time": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"end_time": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"arn": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsAppautoscalingScheduledActionCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).appautoscalingconn
+
+	input := &applicationautoscaling.PutScheduledActionInput{
+		ScheduledActionName: aws.String(d.Get("name").(string)),
+		ServiceNamespace:    aws.String(d.Get("service_namespace").(string)),
+		ResourceId:          aws.String(d.Get("resource_id").(string)),
+	}
+	if v, ok := d.GetOk("scalable_dimension"); ok {
+		input.ScalableDimension = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("schedule"); ok {
+		input.Schedule = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("scalable_target_action"); ok {
+		sta := &applicationautoscaling.ScalableTargetAction{}
+		raw := v.([]interface{})[0].(map[string]interface{})
+		if max, ok := raw["max_capacity"]; ok {
+			sta.MaxCapacity = aws.Int64(int64(max.(int)))
+		}
+		if min, ok := raw["min_capacity"]; ok {
+			sta.MaxCapacity = aws.Int64(int64(min.(int)))
+		}
+		input.ScalableTargetAction = sta
+	}
+	if v, ok := d.GetOk("start_time"); ok {
+		t, err := time.Parse(awsAppautoscalingScheduleTimeLayout, v.(string))
+		if err != nil {
+			return fmt.Errorf("Error Parsing Appautoscaling Scheduled Action Start Time: %s", err.Error())
+		}
+		input.StartTime = aws.Time(t)
+	}
+	if v, ok := d.GetOk("end_time"); ok {
+		t, err := time.Parse(awsAppautoscalingScheduleTimeLayout, v.(string))
+		if err != nil {
+			return fmt.Errorf("Error Parsing Appautoscaling Scheduled Action End Time: %s", err.Error())
+		}
+		input.EndTime = aws.Time(t)
+	}
+
+	err := resource.Retry(5*time.Minute, func() *resource.RetryError {
+		_, err := conn.PutScheduledAction(input)
+		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				switch aerr.Code() {
+				case applicationautoscaling.ErrCodeObjectNotFoundException:
+					return resource.RetryableError(err)
+				default:
+					return resource.NonRetryableError(err)
+				}
+			}
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(d.Get("name").(string))
```

Comment:
Shouldn't the ID consist of name + service namespace as that's how we look it up? In other words my question is: Can there be more than 1 scheduled action of the same name? If not then ignore me.

Reply:
According to `DeleteScheduledActionInput` docs, `ResourceId, ScheduledActionName, ServiceNamespace` are required so we have to generate id from these three.
Should I convert to hash?

Reply:
I'd suggest we make it something like this:

```go
d.SetId(name+"-"+serviceNamespace+"-"+resourceId)
```
assuming the underscore has no special meaning in either of these fields.